src/jobs/cleanup.js

```diff
@@ -106,10 +171,27 @@ export default function (agenda) {
 		async function processCallback(blobs) {
 			return Promise.all(blobs.map(async ({id}) => {
 				try {
-					if (method === 'deleteBlob') {
+					if (method === 'deleteBlob' || method === 'reQueueBlob') {
 						await channel.deleteQueue(id);
 					}
 
+					if (method === 'reQueueBlob') {
+						const docker = new Docker();
+						const containers = await docker.listContainers({
+							filters: {
+								label: [
+									'fi.nationallibrary.melinda.record-import.container-type=transform-task',
+									`blobId=${id}`
+								]
+							}
+						});
+
+						if (containers.length === 0) {
+							logger.log('warn', `Blob ${id} has no transformer alive. Setting state to PENDING_TRANSFORMATION`);
+							client.updateState({id, state: BLOB_STATE.PENDING_TRANSFORMATION});
```

Comment:
The promise needs to be returned here. Otherwise will proceed to next client call below.